tests/ui-toml/toml_disallowed_type/clippy.toml

```diff
@@ -6,4 +6,5 @@ disallowed-types = [
     "std::thread::Thread",
     "std::time::Instant",
     "std::io::Read",
+    "std::primitive::usize"
```

Comment:
Is this required with the currently implementation, or could one just write `"usize"` here?

Reply:
Agreed, that would be better. The docs for std::primitive says it is re-exports of primitives. The simple names are the canonical names.

Reply:
The problem is that [path_to_res](https://github.com/rust-lang/rust-clippy/blob/master/clippy_utils/src/lib.rs#L466) doesn't allow non `crate::name` paths, and looking at the code I'm not sure it could be altered to work either. Is there a way to go from `Symbol` to `DefId` or search all items visible to the current crate?

Reply:
You can use `PrimTy::from_name`.